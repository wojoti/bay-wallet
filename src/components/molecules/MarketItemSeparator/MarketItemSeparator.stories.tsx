import {ComponentMeta, ComponentStory} from '@storybook/react-native';
import {MarketItemSeparator} from 'components/molecules';

export default {
  title: 'Molecules/MarketItemSeparator',
  component: MarketItemSeparator,
} as ComponentMeta<typeof MarketItemSeparator>;
const Template: ComponentStory<typeof MarketItemSeparator> = args => (
  <MarketItemSeparator {...args} />
);

export const Example = Template.bind({});
Example.args = {
  testId: '',
};
