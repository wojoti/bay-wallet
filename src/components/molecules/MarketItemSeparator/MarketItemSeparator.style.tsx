import {StyleSheet} from 'react-native';
import {separator} from 'theme/colors';

const style = StyleSheet.create({
  wrapper: {
    height: 1,
    backgroundColor: separator,
    width: '85%',
  },
});
export default style;
