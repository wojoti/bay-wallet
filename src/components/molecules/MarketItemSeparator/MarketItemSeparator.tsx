import {Container} from 'components/atoms';
import {PropsWithChildren} from 'react';
import style from './MarketItemSeparator.style';

export type MarketItemSeparatorProps = PropsWithChildren<{
  testId?: string;
}>;

const MarketItemSeparator = ({testId}: MarketItemSeparatorProps) => {
  return (
    <Container testId={testId} justify="center" align="center">
      <Container customStyle={style.wrapper} />
    </Container>
  );
};

export default MarketItemSeparator;
