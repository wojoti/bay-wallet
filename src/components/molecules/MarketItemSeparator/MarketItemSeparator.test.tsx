import '@testing-library/jest-native/extend-expect';
import {render, screen} from '@testing-library/react-native';
import TestRenderer from 'react-test-renderer';
import MarketItemSeparator, {
  MarketItemSeparatorProps,
} from './MarketItemSeparator';

const props: MarketItemSeparatorProps = {
  testId: 'test-marketitemseparator-id',
};

test('should match snapshot', () => {
  const tree = TestRenderer.create(<MarketItemSeparator {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('should render MarketItemSeparator', () => {
  render(<MarketItemSeparator {...props} />);
  const MarketItemSeparatorElement = screen.getByTestId(
    'test-marketitemseparator-id',
  );
  expect(MarketItemSeparatorElement).toBeOnTheScreen();
});
