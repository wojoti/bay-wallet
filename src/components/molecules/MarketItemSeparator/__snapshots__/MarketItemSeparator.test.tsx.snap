// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flex": undefined,
        "flexDirection": "column",
        "justifyContent": "center",
        "width": "100%",
      },
      undefined,
    ]
  }
  testID="test-marketitemseparator-id"
>
  <View
    style={
      [
        {
          "alignItems": "stretch",
          "flex": undefined,
          "flexDirection": "column",
          "justifyContent": "flex-start",
          "width": "100%",
        },
        {
          "backgroundColor": "#F6F6F6",
          "height": 1,
          "width": "85%",
        },
      ]
    }
  />
</View>
`;
